src/common/libflux/message.c

```diff
@@ -523,20 +537,22 @@ flux_msg_t *flux_msg_decode (const void *buf, size_t size)
             errno = EINVAL;
             goto error;
         }
-        if (!(zf = zframe_new (p, n)))
-            goto nomem;
-        if (zmsg_append (zmsg, &zf) < 0)
+
+        if (!(iov = realloc (iov, (iovcnt + 1) * sizeof (struct msg_iovec))))
             goto nomem;
```

Comment:
`realloc()` already sets errno

Also this code leaks iov if the `realloc()` fails.  Assign result to tmp, check for NULL, then assign, like the other realloc in this commit.

Reply:
doh!  that's bad of me.  I remember wanting to fix this line, but i guess i forgot.